File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = []
    di = {}
    
    for i in stages:
        di[i] = di.get(i, 0) + 1
    
    total = len(stages)
    result = []
    for i in range(1, N + 1):
        if total == 0:
            result.append([0, i])
        else:
            result.append([di.get(i, 0) / total, i])
            total -= di.get(i, 0)
    
    result.sort(key = lambda x: (-x[0], x[1]))

    answer = [x[1] for x in result]
    
    return answer

File: 프로그래머스/1/42889. 실패율/실패율.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<pair<double, int>> tmp;

bool cmp(pair<double, int> p1, pair<double, int> p2) {
    return p1.first == p2.first ? p1.second < p2.second : p1.first > p2.first;
}

vector<int> solution(int N, vector<int> stages) {
    vector<int> answer;
    int l = stages.size();
    
    for (int i = 1; i <= N; i++) {
        double a = 0, b =0;
        for (int j = 0; j < l; j++) {
            if (stages[j] == i) {
                a++;
            }
            if (stages[j] >= i) {
                b++;
            }
        }
        if (b != 0) {
            tmp.push_back({a/b, i});
        } else {
            tmp.push_back({0, i});
        }
    }
    
    sort(tmp.begin(), tmp.end(), cmp);
    
    for (int i = 0; i < tmp.size(); i++) {
        answer.push_back(tmp[i].second);
    }
    
    return answer;
}